File: game_manager.py
"""
Game Manager - Core game logic and state management
"""

import pygame
from enum import Enum
from player import Player


class GameState(Enum):
    """Game state enumeration"""
    MENU = "menu"
    LOADING = "loading"
    PLAYING = "playing"
    PAUSED = "paused"
    GAME_OVER = "game_over"


class GameManager:
    """Main game manager - handles game loop and state"""
    
    def __init__(self):
        self.width = 1280
        self.height = 720
        self.screen = pygame.display.set_mode((self.width, self.height))
        pygame.display.set_caption("Make It Rain - Heist Game")
        
        self.clock = pygame.time.Clock()
        self.running = True
        self.state = GameState.PLAYING
        self.fps = 60
        
        # Game objects
        self.player = Player(self.width / 2, self.height / 2)
        
        # Font for debug text
        self.font = pygame.font.Font(None, 24)
    
    def handle_input(self):
        """Handle player input"""
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                self.running = False
            elif event.type == pygame.KEYDOWN:
                if event.key == pygame.K_ESCAPE:
                    self.running = False
                elif event.key == pygame.K_p:
                    # Toggle pause
                    if self.state == GameState.PLAYING:
                        self.state = GameState.PAUSED
                    elif self.state == GameState.PAUSED:
                        self.state = GameState.PLAYING
        
        # Continuous key input for movement
        keys = pygame.key.get_pressed()
        self.player.handle_input(keys)
    
    def update(self, delta_time):
        """Update game logic"""
        if self.state == GameState.PLAYING:
            self.player.update(delta_time, self.width, self.height)
    
    def draw(self):
        """Render everything"""
        # Clear screen
        self.screen.fill((0, 0, 0))
        
        # Draw game objects
        self.player.draw(self.screen)
        
        # Draw debug info
        self._draw_debug_info()
        
        # Update display
        pygame.display.flip()
    
    def _draw_debug_info(self):
        """Draw debug information on screen"""
        debug_texts = [
            f"Make It Rain - Heist Game",
            f"FPS: {self.clock.get_fps():.1f}",
            f"Player Pos: ({self.player.x:.1f}, {self.player.y:.1f})",
            f"State: {self.state.value.upper()}",
            f"Controls: WASD=Move | P=Pause | ESC=Exit"
        ]
        
        for i, text in enumerate(debug_texts):
            surface = self.font.render(text, True, (0, 255, 0))
            self.screen.blit(surface, (10, 10 + i * 25))
    
    def run(self):
        """Main game loop"""
        while self.running:
            delta_time = self.clock.tick(self.fps) / 1000.0  # Convert to seconds
            
            self.handle_input()
            self.update(delta_time)
            self.draw()
        
        print("Game closed. Thanks for playing!")

File: Assets/Scripts/Core/GameManager.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;

namespace MIR.Core
{
    /// <summary>
    /// Main game manager - handles core game logic and state.
    /// </summary>
    public class GameManager
    {
        private Game game;
        private GameState currentState;
        private Vector2 playerPosition;
        private Vector2 playerVelocity;
        private const float PLAYER_SPEED = 200f;

        public enum GameState
        {
            Menu,
            Loading,
            Playing,
            Paused,
            GameOver
        }

        public GameManager(Game game)
        {
            this.game = game;
            currentState = GameState.Menu;
            playerPosition = new Vector2(640, 360);
            playerVelocity = Vector2.Zero;
        }

        public void LoadContent()
        {
            System.Console.WriteLine("Make It Rain - Loading content...");
        }

        public void Update(GameTime gameTime)
        {
            HandleInput(gameTime);
            UpdateGameState(gameTime);
        }

        private void HandleInput(GameTime gameTime)
        {
            KeyboardState keyboardState = Keyboard.GetState();

            // Player movement
            playerVelocity = Vector2.Zero;

            if (keyboardState.IsKeyDown(Keys.W))
                playerVelocity.Y -= PLAYER_SPEED;
            if (keyboardState.IsKeyDown(Keys.S))
                playerVelocity.Y += PLAYER_SPEED;
            if (keyboardState.IsKeyDown(Keys.A))
                playerVelocity.X -= PLAYER_SPEED;
            if (keyboardState.IsKeyDown(Keys.D))
                playerVelocity.X += PLAYER_SPEED;

            // Update position
            playerPosition += playerVelocity * (float)gameTime.ElapsedGameTime.TotalSeconds;

            // Clamp to screen bounds
            playerPosition.X = MathHelper.Clamp(playerPosition.X, 0, 1280 - 32);
            playerPosition.Y = MathHelper.Clamp(playerPosition.Y, 0, 720 - 32);

            // Pause with P
            if (keyboardState.IsKeyDown(Keys.P) && currentState == GameState.Playing)
                currentState = GameState.Paused;
            else if (keyboardState.IsKeyDown(Keys.P) && currentState == GameState.Paused)
                currentState = GameState.Playing;
        }

        private void UpdateGameState(GameTime gameTime)
        {
            switch (currentState)
            {
                case GameState.Menu:
                    currentState = GameState.Playing;
                    break;

                case GameState.Playing:
                    // Game logic here
                    break;

                case GameState.Paused:
                    // Pause logic
                    break;
            }
        }

        public void Draw(SpriteBatch spriteBatch, GameTime gameTime)
        {
            // Draw background
            spriteBatch.DrawString(null, "Make It Rain", new Vector2(10, 10), Color.White);
            spriteBatch.DrawString(null, $"Position: {playerPosition}", new Vector2(10, 30), Color.LimeGreen);
            spriteBatch.DrawString(null, $"State: {currentState}", new Vector2(10, 50), Color.Yellow);
            spriteBatch.DrawString(null, "WASD: Move | P: Pause | ESC: Exit", new Vector2(10, 70), Color.Cyan);

            // Draw player (simple white rectangle)
            spriteBatch.Draw(
                new Texture2D(spriteBatch.GraphicsDevice, 1, 1),
                new Rectangle((int)playerPosition.X, (int)playerPosition.Y, 32, 32),
                Color.LimeGreen
            );
        }
    }
}
